### DISCUSSION TITLE
Zod and Angular template types

### DISCUSSION THREAD

**Discussion by ivan-janssens-de-varebeke-lemon** [colinhacks/zod#560] [Q&A]
I have encountered an issue.
I use Zod in combination with angular (12).
My code was not compiling due to an incorrect type.
Here is a small simplified example of the used code:

```
@Component({...})
export class test {
  tests: testObject[] = [];
}

const testObject = z.enum(['test-1', 'test-2', 'test-3']);
type testObject = z.infer<typeof testObject>
```

In the template I use an `*ngFor` loop. In this loop the type is not correctly passed on as can be seen in following screenshots.

<img width="615" alt="Screenshot 2021-07-28 at 08 41 39" src="https://user-images.githubusercontent.com/56079824/127276181-65aaf020-1021-46c2-befa-c89bdcc1a672.png">

<img width="615" alt="Screenshot 2021-07-28 at 08 41 54" src="https://user-images.githubusercontent.com/56079824/127276206-3b2e272d-f11e-4647-8c22-6c21d7532a9f.png">

When changing the testObject to:
```
const testObject = z.literal('test-1')
  .or(z.literal('test-2'))
  .or(z.literal('test-3'))
```
the type is correctly handled in the template.

Can somebody explain why this is happening?

**Comment by scotttrinh** [comment]
Seems to work as expected on the [TypeScript playground](https://www.typescriptlang.org/play?ts=4.2.3#code/JYWwDg9gTgLgBAbzgLzgXzgMyhEcDkyEAJvgNwBQFAxhAHYDO8MApkwPIBGAVi9fAF4UAOhZ0AriAAUAbXysmAWgCM+ADQEFMRQCZ1mttoDM+ALoBKSjACeYFnC1de-OEOTDgdTCygAeG3YQmA6GTnwwAHw09EwhTAwAXHEwYfwypq5w6ZSY0HBStIzMcEHJDOaIFHDVcBRoQA). What version of TypeScript are you using?

**Reply by ivan-janssens-de-varebeke-lemon** [reply]
The issue is specifically in combination with the angular template and `*ngFor` loop.
This is probably related to angular, but I was hoping somebody here might know the reason
